### ISSUE TITLE
Error with sharing 

### ISSUE THREAD

**Issue by zeeshanqaswar1221** [yasirkula/UnityNativeShare#152]
![image](https://user-images.githubusercontent.com/105287838/195060612-08028f65-009a-4ffe-9fd4-58c1f949606d.png)

I am using this block of code to share my json file on Ipad. When I am on the main dialogue Only one json file with a name is being appeared and I am also checking the number of files I have in files list while doing this
![image](https://user-images.githubusercontent.com/105287838/195061613-22a034d5-2535-4388-a681-7e930cea61ce.png)

But the problem is when I press "Save to Files" It adds an additional file of "text.txt" so as result it saves two files on the drive instead of just my exported json file.

I am confused why is this happening. To test my case what I did was I tried to mail and copy the file and it was not adding any extra files Instead I was getting the only file that I was wanted which is the desired result.

**Comment by yasirkula** [comment]
Is it possible that the file is renamed to text.txt and the other json file was already there before?

**Comment by zeeshanqaswar1221** [comment]
My code is quite straight forward and I personally don't see it happening at the Unity level. I am not sure about whether its happening on the plugin side or not.

**Comment by yasirkula** [comment]
I've seen entries mentioning the saved file automatically being renamed to text.txt but never seen any entries where text.txt was duplicated. Please delete both text.txt and the other json file and try again.

**Comment by zeeshanqaswar1221** [comment]
Can you tell me how can I verify on my side like where it's being renamed to text.txt?

**Comment by yasirkula** [comment]
> Can you tell me how can I verify on my side like where it's being renamed to text.txt?

You can verify it by deleting txt and json from the folder and saving to that folder again. If only txt exists, then the file is renamed.

**Comment by zeeshanqaswar1221** [comment]
there is only one file in the folder I Debuged the path of file when pressing the AddFile Event in Xcode and its only one call to addFile and one debug

**Comment by zeeshanqaswar1221** [comment]
No! I totally understand you Point when I try to click "Save to Files" button at the time "X Folder " which I am pointing to is totally empty. When you press "Save to Files" it shows me that on top "2 Items" and I select the X folder below.

**Comment by zeeshanqaswar1221** [comment]
and I only sent one call to Add a file in Unity which I debugged in Xcode and is precisely only one call and the path I use in AddFile() method is exactly the path to JSON file. What I am confused about is where did this "text.txt" file come from When I am not even creating this file. Is this something happening at plugin level?

**Comment by zeeshanqaswar1221** [comment]
Thanks after So many hits and trials 
![195060612-08028f65-009a-4ffe-9fd4-58c1f949606d](https://user-images.githubusercontent.com/105287838/195603977-713fc518-dfb3-4b5d-b082-8866e7b7bb8c.png)

I removed these two lines and my problem is solved now its not adding any additional text.txt file and It was happening at the plugin level. Between Thanks for your Support.

**Comment by yasirkula** [comment]
Thank you for letting me know about this. I believe you can use SetCallback without any issues. This must be related to SetSubject only. I'll research some more to see how it can be resolved.

**Comment by yasirkula** [comment]
Can you change [these lines](https://github.com/yasirkula/UnityNativeShare/blob/bfb8b4e8124ad52712f0d40d2ffa9b966d03bdbd/Plugins/NativeShare/iOS/NativeShare.mm#L21-L34) with the following code (you can do that in Xcode by modifying NativeShare.mm for testing purposes) and see if it resolves the issue:

```objc
- (id)activityViewController:(UIActivityViewController *)activityViewController itemForActivityType:(NSString *)activityType
{
	return [activityType containsString:@"com.apple.CloudDocsUI.AddToiCloudDrive"] ? nil : [self body];
}

- (NSString *)activityViewController:(UIActivityViewController *)activityViewController subjectForActivityType:(NSString *)activityType
{
	return [activityType containsString:@"com.apple.CloudDocsUI.AddToiCloudDrive"] ? nil : [self subject];
}
```
